src/or/voting_schedule.c

```diff
@@ -138,14 +142,32 @@ voting_schedule_t voting_schedule;
 time_t
 voting_schedule_get_next_valid_after_time(void)
 {
+  bool need_to_recalculate_voting_schedule = false;
+
   /* This is a safe guard in order to make sure that the voting schedule
    * static object is at least initialized. Using this function with a zeroed
    * voting schedule can lead to bugs. */
   if (tor_mem_is_zero((const char *) &voting_schedule,
                       sizeof(voting_schedule))) {
-    voting_schedule_recalculate_timing(get_options(), time(NULL));
+    need_to_recalculate_voting_schedule = true;
+  }
+
+  /* Also make sure we are not using an outdated voting schedule. If we have a
+   * newer consensus, make sure we recalculate the voting schedule. */
+  networkstatus_t *consensus = networkstatus_get_live_consensus(approx_time());
```

Comment:
I would `const` that if possible.

Also, `approx_time()` is used twice in this function so probably a `time_t now` is desirable?

Reply:
Done in 222c5cf5.

Reply:
ACK